### PR TITLE
Assert number of messages in test_messages_documentation.py

### PR DESCRIPTION
**Pull request by matusvalo**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|     | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

### CODE REVIEW

File: doc/test_messages_documentation.py [hunk 1]
```diff
@@ -69,6 +80,12 @@ def __init__(self, test_file: Tuple[str, Path]) -> None:
     def runTest(self) -> None:
         self._runTest()
 
+    def is_good_test_file(self):
```
Comment (DanielNoord) [inline doc/test_messages_documentation.py:83]:
```suggestion
    def is_good_test_file(self) -> bool:
```

File: doc/test_messages_documentation.py [hunk 2]
```diff
@@ -69,6 +80,12 @@ def __init__(self, test_file: Tuple[str, Path]) -> None:
...
+        return self._test_file[1].name == "good.py"
+
+    def is_bad_test_file(self):
```
Comment (DanielNoord) [inline doc/test_messages_documentation.py:86]:
```suggestion
    def is_bad_test_file(self) -> bool:
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: doc/test_messages_documentation.py [hunk 3]
```diff
@@ -4,7 +4,18 @@
 
 """Functional tests for the code examples in the messages documentation."""
 
-from collections import Counter
+import sys
+
+if sys.version_info.major > 3 and sys.version_info.minor > 9:
```
Comment (DanielNoord) [inline doc/test_messages_documentation.py:9]:
```suggestion
if sys.version_info[:2] >= (3, 9)
```

File: doc/test_messages_documentation.py [hunk 4]
```diff
@@ -108,6 +125,10 @@ def _runTest(self) -> None:
         self._linter.check([str(self._test_file[1])])
         expected_messages = self._get_expected()
         actual_messages = self._get_actual()
+        if self.is_good_test_file():
+            assert actual_messages.total() == 0  # type: ignore[attr-defined]
```
Comment (DanielNoord) [inline doc/test_messages_documentation.py:129]: I think it is because `MessageCounter` is still `Counter` without the `totals`. But this is fine for me!

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

**Comment by Pierre-Sassoulas** [comment]
Thank you @matusvalo !

**Pull request merged**